enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -45,43 +66,62 @@ Organizations desire to reuse their existing enterprise log solutions to store c
 The specific goals of this proposal are:
 
 * Provide an API that identifies log sources and their desired forwarding outputs.
-* Send logs to an Elasticsearch cluster not managed by the cluster logging infrastructure
-* Send logs to a fluentd instance not managed by the cluster logging infrastructure via fluentd's `forward` protocol
+* Send logs to endpoints not managed by the cluster logging infrastructure:
+  * An Elasticsearch cluster
+  * An endpoint that accepts the `fluent_forward` protocol (fluentd, fluentbit, others...).
+  * An endpoint that accepts syslog via UDP, TCP or TLS.
+  * Others <!-- XXX(alanconway) what is the target set for GA? -->
+* Provide an extensible API so that will allow new output types to be added in future.
 * Allow deployment of log forwarding without deploying the entirety of the cluster logging infrastructure (e.g. Kibana, Elasticsearch)
 * Support TLS between the collector and destination if so configured
 * Use the same log forwarding logic to write logs to a specified output or to the default storage managed by cluster logging (e.g. Elasticsearch).
+* Deploy global forwarders that can forward logs from any container in the cluster.
+* Deploy namespace-restricted forwarders that can only forward logs from their namespace.
+* Deploy multiple forwarder instances with separate RBAC access controls.
```

Comment:
how will the rbac be distinguished?  since it's a single CRD, the only way to really control who can create/edit them is via standard rbac rules "user x can create/edit/view resource y in namespace z" (with the potential additional restriction of "only when the resource is named foo").

if that's all we're talking about, i dunno that we need to list it here, it's implied by the prior statements about having multiple forwarder configuration instances.

Reply:
Agreed. This goes to the requirement of having 2 distinct admin roles: each can create its own instance, but they can't modify each others'. This has dropped out for GA, we are looking for other possible workarounds, so this may or may not be how it works eventually.